ZenPacks/training/OpenWeather/dsplugins.py

```diff
@@ -37,47 +39,91 @@ def config_key(cls, datasource, context):
+            'api_version': context.zOWeatherAPIVersion,
             'city_name': context.city_name,
             'country_code': context.country_code,
             'location_name': context.title,
             }
 
+    def camel_case_to_snake(self, name):
+        s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
+        return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+
     @inlineCallbacks
     def collect(self, config):
         data = self.new_data()
 
         for datasource in config.datasources:
             try:
                 response = yield getPage(
-                    'https://api.openweathermap.org/data/2.5/weather?q={city_name},{country_code}&appid={api_key}'
+                    'https://api.openweathermap.org/data/{api_version}/weather?q={city_name},{country_code}&units=metric&appid={api_key}'
                     .format(
                         api_key=datasource.params['api_key'],
+                        api_version=datasource.params['api_version'],
                         city_name=datasource.params['city_name'],
                         country_code=datasource.params['country_code']))
 
                 response = json.loads(response)
-            except Exception:
+            except Exception, e:
                 LOG.exception(
                     "%s: failed to get conditions data for %s",
                     config.id,
                     datasource.params['location_name'])
 
+                data['events'].append({
+                    'device': config.id,
+                    'component': datasource.component,
+                    'eventKey': 'oweather-conditions',
+                    'eventClassKey': 'oweather',
+                    'severity': SEVERITY_ERROR,
+                    'summary':
+                        '{}: failed to get conditions data for {}'
+                        .format(config.id, datasource.params['location_name']),
+                    'message':
+                        '{}: failed to get conditions data for {}. Reason: {}'
+                        .format(config.id, datasource.params['location_name'], e.message),
+                })
                 continue
 
-            weather = response['weather'][0]
-            main = response['main']
-            wind = response['wind']
+            try:
+                weather = response['weather'][0]
+                main = response['main']
+                wind = response['wind']
+            except (KeyError, TypeError):
+                LOG.exception(
+                    "%s: failed to get conditions data for %s",
+                    config.id,
+                    datasource.params['location_name'])
+
+                data['events'].append({
+                    'device': config.id,
+                    'component': datasource.component,
+                    'eventKey': 'oweather-conditions',
+                    'eventClassKey': 'oweather',
+                    'severity': SEVERITY_ERROR,
+                    'message':
```

Comment:
summary=message. Specify summary
Please clear event
Generate different event for specific case